src/main/java/com/ocient/jdbc/XGConnection.java

```
@@ -944,10 +945,14 @@ private void saveSecondaryInterfaces(final List<String> newCmdComps, final List<
 				final StringTokenizer tokens = new StringTokenizer(connInfo, ":", false);
 				final String connHost = tokens.nextToken();
 				final int connPort = Integer.parseInt(tokens.nextToken());
-				final InetAddress[] addrs = InetAddress.getAllByName(connHost);
-				for(final InetAddress addr : addrs){
-					connInfo = addr.toString().substring(addr.toString().indexOf('/') + 1) + ":" + connPort;
-					this.secondaryInterfaces.get(i).add(connInfo);
+				try {
+					final InetAddress[] addrs = InetAddress.getAllByName(connHost);
+					for (final InetAddress addr : addrs) {
+						connInfo = addr.toString().substring(addr.toString().indexOf('/') + 1) + ":" + connPort;
+						this.secondaryInterfaces.get(i).add(connInfo);
+					}
+				} catch (final UnknownHostException e) {
+					LOGGER.log(Level.DEBUG, "Cannot resolve hostname %s received from database", connHost);
```

LGTM, I see that we swallow this exception in pyocient as well.

One question: do we need an array bounds check when the now potentially-empty list is consumed? I can't tell for certain, but it looks like this code assumes all of the list elements in `XGConnection#secondaryInterfaces` are of equal length. 
https://github.com/Xeograph/ocient_jdbc/blob/78ca3419144aafa1ffd557de7d082beb07de4f5e/src/main/java/com/ocient/jdbc/XGConnection.java#L3316-L3322

---

this whole thing is fragile, but if names for one mode (SQL0, say) are resolvable, but names for another  node (SQL!) are not, this is just going to be screwed in any case